src/router.rs

```diff
@@ -934,6 +934,26 @@ mod tests {
         let info = re.match_with_params(&req, 0).unwrap();
         assert_eq!(info.get("version").unwrap(), "151");
         assert_eq!(info.get("id").unwrap(), "adahg32");
+
+        let re = ResourceDef::new("/v{version:[0-9]+}/resource/{id}");
+        assert!(re.is_match("/v1/resource/320120"));
+        assert!(!re.is_match("/v/resource/1"));
+        assert!(!re.is_match("/va/resource/1"));
+
+        let req = TestRequest::with_uri("/v151/resource/adahg32").finish();
+        let info = re.match_with_params(&req, 0).unwrap();
+        assert_eq!(info.get("version").unwrap(), "151");
+        assert_eq!(info.get("id").unwrap(), "adahg32");
+
+        let re = ResourceDef::new("/v{version:(?:[0-9])+}/resource/{id}");
+        assert!(re.is_match("/v1/resource/320120"));
+        assert!(!re.is_match("/v/resource/1"));
+        assert!(!re.is_match("/va/resource/1"));
+
+        let req = TestRequest::with_uri("/v151/resource/adahg32").finish();
+        let info = re.match_with_params(&req, 0).unwrap();
+        assert_eq!(info.get("version").unwrap(), "151");
```

Comment:
Sorry, I missed the below assertion.

Looking at the resource first capture `v{version:(?:[0-9])+}` shouldn't it prevent capturing at all?
If I'm not mistaken `(?:.+)` should not capture anything and current behavior might be misleading?

Reply:
This is actually a very bad example to talk about as it doesn't really make sense (at least not for my use case).
It makes more sense to talk about something like this: ```[a-z0-9]+(?:[._-][a-z0-9]+)*```, where the non-capturing group makes sense, because the group can be repeated

Reply:
The thing is that it may be unexpected for a user to see the non-capturing pattern to actually capture something.
So I think it would be better if we would be consistent with regex behavior. Though I do imagine the usage of non-capturing groups would be quite rare.

Reply:
Good point, it can easily be solved in a different way. I'll close this PR